radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/messaging/MessageCentralImpl.java

```diff
@@ -50,93 +50,80 @@
 
-	private static final MessageListenerList EMPTY_MESSAGE_LISTENER_LIST = new MessageListenerList();
-
 	// Dependencies
-	private final Serialization serialization;
 	private final TransportManager connectionManager;
 	private final AddressBook addressBook;
 	private final SystemCounters counters;
 
-	// Local data
-
 	// Message dispatching
 	private final MessageDispatcher messageDispatcher;
-
-	// Listeners
-	private final ConcurrentHashMap<Class<? extends Message>, MessageListenerList> listeners = new ConcurrentHashMap<>();
+	private final MessagePreprocessor messagePreprocessor;
 
 	// Our time base for System.nanoTime() differences.  Per documentation can only compare deltas
 	private final long timeBase = System.nanoTime();
 
 	// Listeners we are managing
 	private final List<Transport> transports;
 
-	// Limit rate at which dropped packet logs are produced
-	private final RateLimiter inboundLogRateLimiter = RateLimiter.create(1.0);
 	private final RateLimiter outboundLogRateLimiter = RateLimiter.create(1.0);
 
 	// Inbound message handling
-	private final SimpleBlockingQueue<InboundMessageEvent> inboundQueue;
-	private final SimpleThreadPool<InboundMessageEvent> inboundThreadPool;
+	private final RoundRobinBackpressuredProcessor<InboundMessage> inboundMessageProcessor =
+		new RoundRobinBackpressuredProcessor<>();
+
+	private final RoundRobinBackpressuredProcessor<Pair<Peer, Message>> peerMessageProcessor =
+		new RoundRobinBackpressuredProcessor<>();
 
 	// Outbound message handling
 	private final SimpleBlockingQueue<OutboundMessageEvent> outboundQueue;
 	private final SimpleThreadPool<OutboundMessageEvent> outboundThreadPool;
 
-
 	@Inject
 	MessageCentralImpl(
 		MessageCentralConfiguration config,
 		Serialization serialization,
 		TransportManager transportManager,
 		AddressBook addressBook,
 		TimeSupplier timeSource,
-		EventQueueFactory<InboundMessageEvent> inboundEventQueueFactory,
 		EventQueueFactory<OutboundMessageEvent> outboundEventQueueFactory,
 		LocalSystem localSystem,
 		SystemCounters counters,
 		Hasher hasher,
 		HashSigner hashSigner
 	) {
 		this.counters = Objects.requireNonNull(counters);
-		this.inboundQueue = inboundEventQueueFactory.createEventQueue(
-			config.messagingInboundQueueMax(8192),
-			InboundMessageEvent.comparator()
-		);
 		this.outboundQueue = outboundEventQueueFactory.createEventQueue(
 			config.messagingOutboundQueueMax(16384),
 			OutboundMessageEvent.comparator()
 		);
 
-		this.serialization = Objects.requireNonNull(serialization);
 		this.connectionManager = Objects.requireNonNull(transportManager);
 		this.addressBook = Objects.requireNonNull(addressBook);
 
 		Objects.requireNonNull(timeSource);
+		Objects.requireNonNull(serialization);
+
 		this.messageDispatcher = new MessageDispatcher(
 			counters,
 			config,
 			serialization,
 			timeSource,
+			hasher,
+			hashSigner
+		);
+
+		this.messagePreprocessor = new MessagePreprocessor(
```

Comment:
fyi: I've split MessageDispatcher into MessageDispatcher (that handles OUT messages) and MessagePreprocessor (that handles IN messages).